src/components/achievements/AchievementItem.js

```diff
@@ -1,39 +1,72 @@
+import moment from 'moment';
 import ProptTypes from 'prop-types';
 import React, { useState } from 'react';
+
+import OpenModal from '~/components/modal/OpenModal';
 import EditOptions from '~/components/editoptions/EditOptions';
+import AddAchievement from '~/components/form/achievement/AddAchievement';
 
-const AchievementItem = ({ title, date, preview, onHiddenIconClicked }) => {
+const AchievementItem = ({ title, date, description, preview, onHiddenIconClicked, onDelete }) => {
   const [hidden, setHidden] = useState(false);
+  const [editAchievement, setEdit] = useState(false);
 
   if (hidden && preview) {
     return <></>;
   }
 
-  const onHiddenBtnClicked = e => {
-    e.preventDefault();
+  const toggleEditAchievement = () => setEdit(!editAchievement);
+
+  const editBtnCloseHandler = () => {
+    setEdit(!editAchievement);
+  };
+
+  const onHiddenBtnClicked = () => {
     setHidden(!hidden);
-    onHiddenIconClicked(e, title);
+    onHiddenIconClicked(title);
+  };
+
+  const deleteIconClickedHandler = () => {
+    onDelete(title, date);
   };
 
   return (
-    <div className="achievements__row">
+    <div className={!hidden ? 'achievements__row' : 'achievements__row achievements--hidden'}>
       <div className="achievements__row-header">
         <div className="sub-title">
           {title}
           {hidden && <span className="hidden-tag">Hidden</span>}
         </div>
-        {!preview && <EditOptions isHidden={hidden} onHiddenIconClicked={onHiddenBtnClicked} />}
+        {!preview && (
+          <EditOptions
+            isHidden={hidden}
+            onHiddenIconClicked={onHiddenBtnClicked}
+            onEditButtonClicked={toggleEditAchievement}
+            onDeleteButtonClicked={deleteIconClickedHandler}
+          />
+        )}
+        {editAchievement && (
+          <OpenModal
+            component={AddAchievement}
+            onClose={editBtnCloseHandler}
+            showModal={editAchievement}
+            isEdit={editAchievement}
+            data={editAchievement ? { name: title, date: date, description: description } : ''}
+          ></OpenModal>
+        )}
       </div>
-      <div className="achievements__year">{date}</div>
+      <div className="year year--dark">{moment(date).format('MMMM YYYY')}</div>
```

Comment:
you can use the formatter in `date/FormatDate.js`

Reply:
done for all similar cases